File: src/main/java/guru/xgm/angelscript/AngelScriptElementFactory.java
package guru.xgm.angelscript;

import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiFileFactory;
import guru.xgm.angelscript.psi.AngelScriptFile;
import guru.xgm.angelscript.psi.AngelScriptIdentifier;

public class AngelScriptElementFactory {
    public static AngelScriptIdentifier createIdentifier(Project project, String name) {
        final AngelScriptFile file = createFile(project, name);
        return (AngelScriptIdentifier) file.getFirstChild();
    }

    public static AngelScriptFile createFile(Project project, String text) {
        String name = "dummy.simple";
        return (AngelScriptFile) PsiFileFactory.getInstance(project).
                createFileFromText(name, AngelScriptFileType.INSTANCE, text);
    }
}

File: src/main/kotlin/raft/war/language/angelscript/psi/AngelScriptElementFactory.kt
package raft.war.language.angelscript.psi

import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiFileFactory
import raft.war.language.angelscript.AngelScriptLanguage.Companion.instance
import raft.war.language.angelscript.psi.file.AngelScriptFile

object AngelScriptElementFactory {
    private fun createFile(project: Project, text: String): AngelScriptFile {
        val name = "dummy.as"
        return PsiFileFactory.getInstance(project).createFileFromText(name, instance, text) as AngelScriptFile
    }

    private fun recreateExpr(project: Project, value: String): AngelScriptExpr {
        val file = createFile(project, "int fuckingCrutch = $value;")
        return (file.firstChild as AngelScriptVar).exprList[0]
    }

    fun replaceExprChild(project: Project, target: PsiElement, value: String) {
        val child = recreateExpr(project, value).firstChild ?: return
        target.replace(child)
    }
}
